packages/backend/src/core/reports/ReportUpdater.ts

```diff
@@ -49,11 +53,20 @@ export class ReportUpdater {
     ])
     this.logger.debug('Prices and balances ready')
     const reports = createReports(prices, balances, this.projects)
-    await this.reportRepository.addOrUpdateMany(reports)
-    await this.reportStatusRepository.add({
-      configHash: this.configHash,
+    addOpTokenReport(reports, prices, timestamp)
+    const aggregatedReports = aggregateReports(
+      reports,
+      this.projects,
       timestamp,
-    })
+    )
+    await Promise.all([
+      this.reportRepository.addOrUpdateMany(reports),
+      this.aggregateReportsRepository.addOrUpdateMany(aggregatedReports),
+      this.reportStatusRepository.add({
```

Comment:
This shouldn't happen in parallel. Only save the report status after the other operations have completed